api/api.go

```diff
@@ -217,10 +233,15 @@ func getMeshStatus(api *API) meshStatus {
 	return strippedStatus
 }
 
-func (api *API) alertGroups(w http.ResponseWriter, req *http.Request) {
-	var err error
-	matchers := []*labels.Matcher{}
-	if filter := req.FormValue("filter"); filter != "" {
+func (api *API) alertGroups(w http.ResponseWriter, r *http.Request) {
```

Comment:
I would prefer to keep this PR as lean as possible without adding unneeded functionality on an endpoint that we want to get rid of anyways. Especially as it not only touches `api.go` but `dispatch.go` and `main.go` as well.